packages/provider-utils/src/get-effective-abort-signal.ts

```diff
@@ -0,0 +1,26 @@
+export function getEffectiveAbortSignal(
+  abortSignal: AbortSignal | undefined,
+  timeout: number | undefined
+): { signal: AbortSignal | undefined; clearTimeout: () => void } {
+  let effectiveAbortSignal = abortSignal;
+  let timeoutId: NodeJS.Timeout | null = null;
```

Comment:
will this create node dependency that leads to issues in browsers

Reply:
fixed